code/game/machinery/doors/airlock.dm

```diff
@@ -1355,6 +1360,11 @@
 		log_combat(user, src, message)
 		add_hiddenprint(user)
 
+/obj/machinery/door/airlock/emp_act(severity)
+	. = ..()
+	if(secondsElectrified == MACHINE_ELECTRIFIED_PERMANENT) // door/emp_act() doesn't call the helper proc that sets diag huds.
```

Comment:
why not change that?

Reply:
set_electrified exists at /airlock level, is there a need to add it to everything that isn't an airlock?